### PR TITLE
[buffers] Add handler for the 'create_only_config_db_buffers' configuration knob

### PR DESCRIPTION
**Pull request by vadymhlushko-mlnx**
**What I did**
Add support in the orchagent for the configuration knob:

```
{
    "DEVICE_METADATA": {
        "localhost": {
            "create_only_config_db_buffers": "true"
            ...
        }
    }
}
```

**Why I did it**
If the `"create_only_config_db_buffers"` is equal to `"true"` - buffers will be created according to the config_db configuration (for example `BUFFER_QUEUE|*` table).
If the `"create_only_config_db_buffers"` is equal to `"false"` or does not exist - the maximum available buffers (which are read from SAI) will be created, regardless of `the config_db` buffer config.

**How I verified it**
Add UT.

Manual verification:
1. Install the image with this PR included on the not `MSFT SKU` switch
2. Check the `show queue counters` output and verify that only configured in `CONFIG_DB` buffers are created
```
root@sonic:/home/admin# show queue counters
     Port    TxQ    Counter/pkts    Counter/bytes    Drop/pkts    Drop/bytes
---------  -----  --------------  ---------------  -----------  ------------
Ethernet0    UC0               0                0            0           N/A
Ethernet0    UC1               0                0            0           N/A
Ethernet0    UC2               0                0            0           N/A
Ethernet0    UC3               0                0            0           N/A
Ethernet0    UC4               0                0            0           N/A
Ethernet0    UC5               0                0            0           N/A
Ethernet0    UC6               0                0            0           N/A
```
3. Open the `/usr/share/sonic/device/$DEVICE/$SKU/create_only_config_db_buffers.json` and change it to:
```
"create_only_config_db_buffers": "false"
```
4. Do `config reload`
5. Check the `show queue counters` output and verify that all available buffers are created
```
root@sonic:/home/admin# show queue counters
     Port    TxQ    Counter/pkts    Counter/bytes    Drop/pkts    Drop/bytes
---------  -----  --------------  ---------------  -----------  ------------
Ethernet0    UC0               0                0            0           N/A
Ethernet0    UC1               0                0            0           N/A
Ethernet0    UC2               0                0            0           N/A
Ethernet0    UC3               0                0            0           N/A
Ethernet0    UC4               0                0            0           N/A
Ethernet0    UC5               0                0            0           N/A
Ethernet0    UC6               0                0            0           N/A
Ethernet0    UC7              60            15346            0           N/A
Ethernet0    MC8             N/A              N/A          N/A           N/A
Ethernet0    MC9             N/A              N/A          N/A           N/A
Ethernet0   MC10             N/A              N/A          N/A           N/A
Ethernet0   MC11             N/A              N/A          N/A           N/A
Ethernet0   MC12             N/A              N/A          N/A           N/A
Ethernet0   MC13             N/A              N/A          N/A           N/A
Ethernet0   MC14             N/A              N/A          N/A           N/A
Ethernet0   MC15             N/A              N/A          N/A           N/A
```

**Details if related**

### CODE REVIEW

**Comment by prsunny** [comment]
@vadymhlushko-mlnx , can you check why the test is failing?

**Comment by vadymhlushko-mlnx** [comment]
@abdosi @prsunny could you please approve and merge?

File: orchagent/flexcounterorch.cpp [hunk 1]
```diff
@@ -328,11 +329,41 @@ bool FlexCounterOrch::bake()
...
+        {
+            if (createOnlyConfigDbBuffersValue == "true")
+            {
+                return true;
+            }
+        }
+    }
+    catch(const std::system_error& e)
+    {
+        SWSS_LOG_ERROR("System error: %s", e.what());
```
Comment (prsunny) [inline orchagent/flexcounterorch.cpp:348]: can you change this to WARNING? for non existent config_db, we don't want an ERR log printed

**Review by prsunny — COMMENTED** [review]
lgtm, minor comment. @abdosi , @prabhataravind , please review/sign-off. 

@neethajohn for viz

File: orchagent/flexcounterorch.cpp (hunk 1 above)
Comment (vadymhlushko-mlnx) [reply orchagent/flexcounterorch.cpp:348]: The error statement in the `catch` block will be printed only if the `deviceMetadataConfigTable.hget()` throws an exception.
If the `create_only_config_db_buffers` do not exist in the `config_db` the `deviceMetadataConfigTable.hget()` returns `false`.

**Review by prabhataravind — APPROVED** [review]
lgtm

**Review by prsunny — APPROVED** [review]

**Comment by vadymhlushko-mlnx** [comment]
@prsunny  could we include this PR in the `202305` branch? 
I've already checked - it is a clean cherry-pick

**Comment by StormLiangMS** [comment]
@vadymhlushko-mlnx have you tested 202305 image with this PR?

@prsunny could we have ADO to track if support cherry pick to 202305?

**Comment by Blueve** [comment]
@StormLiangMS Please use this ADO: 25518393

**Pull request merged**